pkg/querier/querier.go

```diff
@@ -204,18 +205,20 @@ func NewQueryable(distributor QueryableWithFilter, stores []QueryableWithFilter,
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
 		now := time.Now()
 
-		if cfg.MaxQueryIntoFuture > 0 {
```

Comment:
Moved to `validateQueryTimeRange()`, which is also called in the `Select()`. If this limit is not enforced in the `Select()` too, we're not really enforcing it. Unit tests were weak: I spotted this issue improving unit tests.

Reply:
Is this because `Select` uses hints?

Reply:
Exactly.